src/error.rs

```diff
@@ -0,0 +1,43 @@
+use std::result::Result as StdResult;
+
+use thiserror::Error as ThisError;
+
+#[allow(missing_docs)]
+pub type Result<T> = StdResult<T, Error>;
+
+#[allow(missing_docs)]
+#[derive(ThisError, Debug)]
+#[error(transparent)]
+pub struct Error(#[from] InternalError);
```

Comment:
This probably doesn't need to exist. Ideally this would just using `http_types::Error` instead:  https://docs.rs/http-types/2.3.0/http_types/struct.Error.html

(It's implemented like this: https://github.com/http-rs/http-types/blob/master/src/error.rs)